tools/mig_agent_cfg.go

```diff
@@ -0,0 +1,274 @@
+    socket           = "127.0.0.1:{{.StatSocketPort}}"
+
+    ; frequency at which heartbeat messages are sent to the MIG relay
+    heartbeatfreq    = "120s"
+
+    ; timeout after which a module that has not finished is killed by the agent
+    moduletimeout    = "300s"
+
+    ; in immortal mode, the agent that encounter a fatal error
+    ; will attempt to restart itself instead of just shutting down
+    isimmortal       = {{.Immortal}}
+
+    ; installservice orders the agent to deploy a service init configuration
+    ; and start itself during the endpoint's boot process
+    installservice   = {{.InstallAsService}}
+
+    ; attempt to retrieve the public IP behind which the agent is running
+    discoverpublicip = on
+
+    ; in check-in mode, the agent connects to the relay, runs all pending commands
+    ; and exits. this mode is used to run the agent as a cron job, not a daemon.
+    checkin = {{.CronMode}}
+
+    refreshenv = "5m"
+
+    ; enable privacy mode
+    extraprivacymode = {{.ExtraPrivacy}}
+
+    ; Tags help investigators to target specific agents for queries.
+    ; Multiple tags can be supplied as demonstrated below and the tag name and value are separated by a colon.
+    ; tags = "operator:example"
+    ; tags = "exampleTag:other"{{ range $tag, $value := .Tags }}
+    tags = "{{ $tag }}:{{ $value }}"{{ end }}
+
+[stats]
+    maxactions = 15
+
+[certs]
+    ca  = "/etc/mig/ca.crt"
+    cert = "/etc/mig/agent.crt"
+    key = "/etc/mig/agent.key"
+
+[logging]
+    mode    = "file" ; stdout | file | syslog
+    level   = "debug"
+    file    = "/var/log/mig-agent.log"
+    maxfilesize = 10485760
+`
+
+// Defaults
+const (
+	defaultAPIURL           = "https://api.mig.mozilla.org/api/v1/"
+	defaultStatSocketPort   = 51664
+	defaultImmortal         = true
+	defaultInstallAsService = true
+	defaultCronMode         = false
+	defaultExtraPrivacy     = true
+)
+
+type proxyList []string
+type tagList []string
+
+// Config contains user-supplied configurable values that will be translated
+// and injected into `release/mig-agent.cfg.template` to produce a valid
+// `mig-agent.cfg`.
+type Config struct {
+	RelayAddress     string
+	APIURL           string
+	Proxies          proxyList
+  Tags             tagList
+	StatSocketPort   uint
+	Immortal         bool
+	InstallAsService bool
+	CronMode         bool
+	ExtraPrivacy     bool
+}
+
+// translatedConfig contains the same data as `Config` with some
+// transformations to make the output of this program consistent
+// with what mig agents expect.
+type translatedConfig struct {
+	RelayAddress     string
+	APIURL           string
+	Proxies          string
+  Tags             map[string]string
+	StatSocketPort   uint16
+	Immortal         string
+	InstallAsService string
+	CronMode         string
+	ExtraPrivacy     string
+}
+
+// onOrOff converts a bool from `true` to `"on"` and `false` to `"off"`.
+func onOrOff(on bool) string {
+	if on {
+		return "on"
+	} else {
+		return "off"
+	}
```

Comment:
The preceding lines of this script have different whitespace than the following lines - 8-space versus 2-space. Consider normalizing to one form only.